internal/venice-common/src/main/java/com/linkedin/venice/blobtransfer/NettyP2PBlobTransferManager.java

```diff
@@ -0,0 +1,72 @@
+package com.linkedin.venice.blobtransfer;
+
+import com.linkedin.venice.blobtransfer.client.NettyFileTransferClient;
+import com.linkedin.venice.blobtransfer.server.P2PBlobTransferService;
+import com.linkedin.venice.exceptions.StorePartitionNotFoundException;
+import com.linkedin.venice.exceptions.VeniceNoStoreException;
+import com.linkedin.venice.exceptions.VenicePeersNotFoundException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * Netty based P2P blob transfer manager implementation.
+ * Upon start, it starts the blob transfer service and the client, so it can receive requests from peers to serve
+ * blobs and in the meanwhile, it can make requests to other peers to fetch blobs.
+ */
+public class NettyP2PBlobTransferManager implements P2PBlobTransferManager<Void> {
+  private static final Logger LOGGER = LogManager.getLogger(NettyP2PBlobTransferManager.class);
+
+  private final P2PBlobTransferService blobTransferService;
+  // netty client is responsible to make requests against other peers for blob fetching
+  protected final NettyFileTransferClient nettyClient;
+  // peer finder is responsible to find the peers that have the requested blob
+  protected final BlobFinder peerFinder;
+
+  public NettyP2PBlobTransferManager(
+      P2PBlobTransferService blobTransferService,
+      NettyFileTransferClient nettyClient,
+      BlobFinder peerFinder) {
+    this.blobTransferService = blobTransferService;
+    this.nettyClient = nettyClient;
+    this.peerFinder = peerFinder;
+  }
+
+  @Override
+  public void start() throws Exception {
+    blobTransferService.start();
+  }
+
+  @Override
+  public CompletionStage<InputStream> get(String storeName, int version, int partition)
+      throws VeniceNoStoreException, VenicePeersNotFoundException {
+    CompletionStage<InputStream> inputStream;
+    BlobPeersDiscoveryResponse response = peerFinder.discoverBlobPeers(storeName, version, partition);
```

Comment:
FYI - this is not getting from the router; although, it doesn't matter in this PR. I can send out a new one